src/aura/BGE_DataImportBatchEntry/BGE_DataImportBatchEntryHelper.js

```diff
@@ -83,6 +83,21 @@
      * @description: invoke the Data Importer on all records associated with this Batch
      */
     processBatch: function(component) {
+        var totals = component.get('v.totals');
+        var record = component.get('v.record');
+        if (record.RequireTotalMatch__c) {
+            if (totals.countGifts == 0 && totals.totalGiftAmount == 0) {
```

Comment:
I think this should be checking against Expected_Total_Batch_Amount__c and Expected_Count_of_Gifts__c, right? Because we'll want to show the error if require total match is true and the expected values are 0.